### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by PierreBarba**
# hello reviewer I hope you are having a great day, any suggestion will be welcome, thank you very much for the support 😄 🌴

### Add RSpec as a gem to your project.

- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
- If response status was correct.
- If a correct template was rendered.
- If the response body includes correct placeholder text.

### CODE REVIEW

**Review by HAMMAS-SALEEM — COMMENTED** [review]
# Invalid Submission :red_circle:

invalid-submission

## Hey @PierreBarba :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements but I cannot proceed review of your pull request because there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

# Mandatory Changes ♻️

- [ ] It would be nice if you remove all the section descriptions, instructions, and unnecessary information from your readme file to make your readme descriptive, clean, and professional 👍
- [ ] It would be nice if you add the necessary commands for running your application on the local machine 👍 such as

### Install
```ruby
bundle install
rails db:create
rails db:migrate
```
### Run tests
```ruby
rspec spec
```

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :request do
+  describe 'Get id' do
+    it 'returns a successful response' do
+      get '/users/:id/posts/:id/'
+      expect(response).to be_successful
+    end
+
```
Comment (BenMukebo) [inline spec/controllers/posts_controller_spec.rb]: - [ ] Great job so far 👏🏻 However the `index` and `show` actions for `User` and `Post` controllers should be tested. Actually, you're missing the `show` actions tests. please add them and make sure to include tests that ensure the show response body includes the correct placeholder text. Example : `expect(response.body).to include("Welcome to user index page")`

```ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @PierreBarba,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   

 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

- [ ] Kindly remove  React.js in the Client Tech Stack  since we're not using them in the project 👇🏻 

image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


- [OPTIONAL ] Use descriptive commit messages.  The commit message should be written in the imperative tense and capitalized. [Read more about how to write descriptive commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8). Example:
  - :x: Wrong: added Descriptive README
  - :heavy_check_mark: Right: Add descriptive README 

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @PierreBarba, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good  implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**